rest-api-spec/src/yamlRestCompatTest/resources/rest-api-spec/api/nodes.hotthreads.json

```diff
@@ -0,0 +1,102 @@
+{
+  "nodes.hotthreads":{
+    "documentation":{
+      "url":"https://www.elastic.co/guide/en/elasticsearch/reference/master/cluster-nodes-hot-threads.html",
+      "description":"Returns information about hot threads on each node in the cluster."
+    },
+    "stability":"stable",
+    "visibility":"public",
+    "headers":{
+      "accept": [ "text/plain"]
+    },
+    "url":{
+      "paths":[
+        {
+          "path":"/_cluster/nodes/hotthreads",
```

Comment:
would it more correct to re-introduce this path (and the like) as deprecated in the spec for 7.x and add the allowed warning (for any tests that call this API) ?

This would allow you to avoid introducing the spec only for compatibility.

Reply:
I like that idea! Just to make sure I'm following -- that'd be a PR pointed at 7.x (no version that hits master) that adds these as deprecated endpoints to the `rest-api-spec`, and then in this PR we wouldn't need a new test or any specs, because this would all execute automatically by way of the `yamlRestCompatTest` tests.

Is there any downside to adding these as deprecated in 7.x? I'm thinking, for example, that perhaps the autogenerated client code might actually start **using them** (if so, that'd be strictly worse, and we probably wouldn't want to go this way).

@pgomulka what do you think?

Reply:
i like that idea too!
if a test client will use it, maybe we can ignore the warning in 7.x for this? Like we do for types removal in 7.x

Reply:
>  that'd be a PR pointed at 7.x (no version that hits master) that adds these as deprecated endpoints to the rest-api-spec, 

yes, that is what i meant. I think it works well in this case (and any other one off cases) since it probably should have been that way from the start and should have minimal pollution in 7.x.

However, I don't think this should be a rule of thumb for all missing paths. For example, for the `_xpack/` prefix i can see the argument for not polluting 7.x since there are so many, somewhat inconsistent (depending on when it was added), and has a lot of associated tests that would need to allow warnings.